File: LEETCODE/LineSweep/1854_maximumPopulationYear/1854_maximumPopulationYear.java
class Solution {
    public int maximumPopulation(int[][] logs) {
        List<int[]> events= new ArrayList<>();
        for(int log[]: logs){
            events.add(new int[]{log[0],+1});
            events.add(new int[]{log[1],-1});
        }
        Collections.sort(events,(a,b)->{
            if(a[0]==b[0]) return a[1]-b[1];
            return a[0]-b[0];
        });
        int currPopulation=0; int maxPopulation=0,minYear=2055;
        for(int[] event: events){
            currPopulation+= event[1];
            if(currPopulation> maxPopulation){
                maxPopulation= currPopulation;
                minYear= event[0];
            }
        }
        return minYear;
        
    }
}

File: LEETCODE/LineSweep/1854_maximumPopulationYear/1854_maximumPopulationYear.py

class Solution:
    def maximumPopulation(self, logs: List[List[int]]) -> int:
        events=[]
        for start,end in logs:
            events.append((start,1))
            events.append((end,-1))
        events.sort(key=lambda x: (x[0],x[1]))
        curr_pop,max_pop,min_year=0,0,2055
        for year,change in events:
            curr_pop+= change
            if curr_pop> max_pop:
                max_pop= curr_pop
                min_year= year
        
        return min_year
